### PR TITLE
It's GANtastic

### PR DESCRIPTION
**Pull request by jg10545**
Changes in this branch:

* combined inpainter and discriminator into a single training step. I was hoping this would reduce oscillation (unclear) but if nothing else is more efficient
* added `Trainer.fit_mask_generator()` and `Trainer.fit_inpainter()` methods so you can independently run the different training phases. My hypothesis that end-to-end training might not matter is definitely wrong- at least on Shipsnet, letting the two co-adapt matters a lot.
* for the inpainter training phase, there's a separate step now to build the mask buffer. this only uses positive images (since masks for negative images should be mostly empty, leading to trivial solutions in inpainter training)
* added learning rate decay

Testing on Shipsnet- LR decay seems to make a big difference.

### CODE REVIEW

**Comment by jg10545** [comment]
Flagging @zamorarr in case you have any thoughts/concerns/ideas/complaints.
